pymdownx/blocks/block.py

```diff
@@ -165,13 +170,13 @@ def type_string_in(accepted, insensitive=True):
     )
 
 
-def type_string_delimiter(split, string_type=type_string):
+def type_string_delimiter(split: str, string_type: Callable[[Any], str] = type_string) -> Callable[[Any], list[str]]:
     """String delimiter function."""
 
     return functools.partial(_delimiter, split=split, string_type=string_type)
 
 
-def type_html_attribute_dict(value):
+def type_html_attribute_dict(value: dict[str, str | list[str]]) -> dict[str, str | list[str]]:
```

Comment:
Many of these accept `Any` for `value`, even this one. They validate and raise an exception if not of the expected type. Technically, you could scope the input to the expected type and have the exception raise if not, or accept Any and ensure the return is as expected. Regardless, we need to be consistent in what we do. Due to the dynamic nature, it may be easiest to just always accept `Any` for value and ensure the return type is as expected.

Reply:
Done